pkg/operator/controller/externaldns/deployment_test.go

```diff
@@ -2259,6 +2261,17 @@ func TestExternalDNSDeploymentChanged(t *testing.T) {
 				testContainer(),
 			}),
 		},
+		{
+			description: "if externalDNS annotation changes",
+			originalDeployment: testDeploymentWithContainers([]corev1.Container{
+				testContainer(),
+			}),
+			mutate: func(dep1 *appsv1.Deployment) {
+				dep1.Annotations = updatedSecretHashAnnotation
+			},
+			expect:             true,
+			expectedDeployment: testDeploymentWithAnnotations(updatedSecretHashAnnotation),
+		},
```

Comment:
Can you please add a test case when the secret annotation didn't exist in the current deployment?

Reply:
test Added